backend/src/test/scala/com/softwaremill/adopttapir/starter/StarterServiceITTest.scala

```diff
@@ -3,49 +3,206 @@ package com.softwaremill.adopttapir.starter
+
+  for { details <- Setup.validConfigurations } {
+    it should s"return zip file containing working sbt folder with: ${details.describe}" in {
+      val backend: SttpBackend[Identity, Any] = HttpURLConnectionBackend()
+
+      // define endpoints integration tests
+      val helloEndpointTest: Option[TestFunction] = Some(port => {
+        val result: String =
+          basicRequest.response(asStringAlways).get(uri"http://localhost:${port}/hello?name=Frodo").send(backend).body
+
+        result shouldBe "Hello Frodo"
+      })
+
+      val docsEndpointTest: Option[TestFunction] = testOption(
+        details.addDocumentation,
+        port => {
+          val result: String =
+            basicRequest.response(asStringAlways).get(uri"http://localhost:${port}/docs/docs.yaml").send(backend).body
+
+          result should include("paths:\n  /hello:")
+        }
+      )
+
+      val metricsEndpointTest: Option[TestFunction] = testOption(
+        details.addMetrics,
+        port => {
+          val result: String =
+            basicRequest.response(asStringAlways).get(uri"http://localhost:${port}/metrics").send(backend).body
+
+          result should (include("# HELP tapir_request_duration_seconds Duration of HTTP requests")
+            and include("# TYPE tapir_request_duration_seconds histogram")
+            and include("# HELP tapir_request_total Total HTTP requests")
+            and include("# TYPE tapir_request_total counter")
+            and include("# HELP tapir_request_active Active HTTP requests")
+            and include("# TYPE tapir_request_active gauge"))
+        }
+      )
+
+      // get implementation for a given configuration, compile it, unit & integration test it
+      val service = ServiceUnderTest(details)
+      service.run(helloEndpointTest, docsEndpointTest, metricsEndpointTest)
     }
   }
 
-  private def createStarterService = {
-    val config = StarterConfig(deleteTempFolder = true, tempPrefix = "sbtService", scalaVersion = "2.13.8")
-    new StarterServiceInTests(config, new ProjectTemplateInTests(config))
+  private def testOption(shouldTest: Boolean, testFunction: TestFunction): Option[TestFunction] = {
+    if (shouldTest) Some(testFunction) else None
   }
+}
 
-  private def sbtCompileTest(file: IO[File]): Unit = {
-    IO.blocking(BFile.newTemporaryDirectory("sbtTesting"))
-      .bracket { tempDir =>
+case class ServiceUnderTest(details: StarterDetails) extends FLogging {
+  import Setup._
+
+  def run(tests: Option[TestFunction]*): Unit = {
+    val log = new StringBuilder()
+    val generatedFile = ZipGenerator.service.generateZipFile(details)
+    IO
+      .blocking(BFile.newTemporaryDirectory("sbtTesting"))
+      .flatMap { tempDir =>
         for {
-          zipFile <- file.map(_.toScala)
+          zipFile <- generatedFile.map(_.toScala)
+          _ <- logStep(log, "* zip file was generated")
           _ <- IO.blocking {
             zipFile.unzipTo(tempDir)
             zipFile.delete()
```

Comment:
I know that this is legacy part but it is worth to do it during this PR, as it should be small.

Generated zip file will be not deleted safely (when any exception in flow occurs). 

```
val generatedFile = ZipGenerator.service.generateZipFile(details)
```

cats `Resource` can be used to properly delete the zip file.

Reply:
done 😄